scripts/saveImage.sh

```diff
@@ -0,0 +1,202 @@
+export DAY_OR_NIGHT="${1}"
+[ "${DAY_OR_NIGHT}" != "DAY" -a "${DAY_OR_NIGHT}" != "NIGHT"  ] && usage_and_exit 1
+
+# ${IMAGE_TO_USE} is the full path to a uniquely-named file created by the capture program.
+# The file name is its final name in the ${ALLSKY_IMAGES}/<date> directory.
+# Because it's a unique name we don't have to worry about another process overwritting it.
+# We modify the file as needed and ultimately save a link to it as ${FULL_FILENAME} since
+# that's what websites look for and what is uploaded.
+
+IMAGE_TO_USE="${2}"
+shift 2
+if [ ! -f "${IMAGE_TO_USE}" ] ; then
+	echo -e "${RED}*** ${ME}: ERROR: File '${IMAGE_TO_USE}' not found; ignoring${NC}"
+	exit 2
+fi
+if [ ! -s "${IMAGE_TO_USE}" ] ; then
+	echo -e "${RED}*** ${ME}: ERROR: File '${IMAGE_TO_USE}' is empty; ignoring${NC}"
+	exit 2
+fi
+
+# The image may be in a memory filesystem, so do all the processing there and
+# leave the image used by the website(s) in that directory.
+IMAGE_NAME=$(basename "${IMAGE_TO_USE}")	# just the file name
+WORKING_DIR=$(dirname "${IMAGE_TO_USE}")	# the directory the image is currently in
+
+# Optional full check for bad images.
+if [ "${REMOVE_BAD_IMAGES}" = "true" ]; then
+	# If the return code is 99, the file was bad and deleted so don't continue.
+	"${ALLSKY_SCRIPTS}/removeBadImages.sh" "${WORKING_DIR}" "${IMAGE_NAME}"
+	# removeBadImages.sh displayed error message and deleted the file.
+	[ $? -eq 99 ] && exit 99
+else
+	# Quick check to make sure the image isn't corrupted.
+	identify "${IMAGE_TO_USE}" >/dev/null 2>&1
+	if [ $? -ne 0 ] ; then
+		echo -e "${RED}*** ${ME}: ERROR: '${IMAGE_TO_USE}' is corrupt; not saving.${NC}"
+		exit 3
+	fi
+fi
+
+# Get passed-in variables.
+# Normally at least the exposure will be passed and the sensor temp if known.
+while [ $# -gt 0 ]; do
+	VARIABLE="THIS_${1%=*}"		# everything before the "="
+	VALUE="${1##*=}"			# everything after the "="
+	shift
+	# Export the variable so other scripts we call can use it.
+	export ${VARIABLE}="${VALUE}"	# need "export" to get indirection to work
+done
+
+# Export so other scripts can use it.
+export CURRENT_IMAGE="${IMAGE_TO_USE}"		# darkCapture.sh and darkSubtract.sh expect CURRENT_IMAGE
+source "${ALLSKY_SCRIPTS}/darkCapture.sh"		# does not return if in darkframe mode
+# xxxx TODO: Dark subtract long-exposure images, even if during daytime.
+# xxxx TODO: Need a config variable to specify the threshold to dark subtract.
+# xxxx TODO: Possibly also for stretching below.
+if [ "${DAY_OR_NIGHT}" = "NIGHT" ] ; then
+	source "${ALLSKY_SCRIPTS}/darkSubtract.sh"	# It will modify the image but not its name.
+fi
+
+# If any of the "convert"s below fail, exit since we won't know if the file was corrupted.
+
+# Resize the image if required
+if [ "${IMG_RESIZE}" = "true" ] ; then
+	[ "${ALLSKY_DEBUG_LEVEL}" -ge 4 ] && echo "${ME}: Resizing '${IMAGE_TO_USE}' to ${IMG_WIDTH}x${IMG_HEIGHT}"
+	convert "${IMAGE_TO_USE}" -resize "${IMG_WIDTH}x${IMG_HEIGHT}" "${IMAGE_TO_USE}"
```

Comment:
We have a problem if all conversions take too much time and we have a short delay between the captures.
In my opinion, as much starlight as possible should be captured. So I want short waiting times. (1s)
All the conversions (resize, crop, annotation, resize for upload, upload) eat up the time, and if another recording comes in between it will crash or malfunction.
Therefore I would like to decouple the images from each other.

Please take a look - works for delays over 0.2s: https://github.com/AndreasLMeg/allsky/blob/6613055fe35c65e30838c98ff89ff073a31c7382/scripts/saveImageNight.sh#L41

Reply:
now I have seen, that tmp directory is used - that solves the problem
is there also a change for web-ui ? no live-view ist displayed. I have to search for the reason....